### ISSUE TITLE
imprecise_flops does incorrectly completely false negative in const contexts

### ISSUE THREAD

**Issue by CommanderStorm** [rust-lang/rust-clippy#17606]
### Summary

when `imprecise_flops` was originally written, there were no floating ops in const context.
Now there are more available (still not all, but some 😄)

Example -> https://doc.rust-lang.org/std/primitive.f64.html#method.mul_add

### Lint Name

imprecise_flops

### Reproducer

https://github.com/rust-lang/rust-clippy/blob/bfc86ca03cbaf26a313589c90d9aaa5704f3280e/tests/ui/floating_point_mul_add.rs#L3-L11

I expected to see this happen:

The lint should also fire, since `mul_add` is const

Instead, this happened:

the lint does not fire, because

https://github.com/rust-lang/rust-clippy/blob/bfc86ca03cbaf26a313589c90d9aaa5704f3280e/clippy_lints/src/floating_point_arithmetic/mod.rs#L110-L113

CC @Gri-ffin since this might be intersting to you and something I noticed during reviewing your PR 😉

Essenitally, I think we should check for this (or a version of it) https://github.com/rust-lang/rust-clippy/blob/e57f468d8f079a7b50627a95121fd16fff8e5bfa/clippy_utils/src/msrvs.rs#L171

### Version

```text
rustc 1.98.0 (88d9e12ae 2026-08-18)
binary: rustc
commit-hash: 88d9e12ae178fab0fb5cc050a94da85685d449ea
commit-date: 2026-08-18
host: x86_64-unknown-linux-gnu
release: 1.98.0
LLVM version: 22.1.8
```
